src/spatialdata_plot/pl/utils.py

```diff
@@ -1444,13 +1478,13 @@ def _validate_colors_element_table_mapping_points_shapes(
                         params.col_for_color.append(col_color)
                         element_table_mapping[element_name] = set()
                     else:
-                        if len(element_table_mapping[element_name].copy()) != 0:
-                            for table_name in element_table_mapping[element_name].copy():
+                        if isinstance(mapping := element_table_mapping[element_name], set) and len(mapping.copy()) != 0:
```

Comment:
reads a bit code-golf-y 👀 Also, why are you copying? Theoretically, you could also remove the `!= 0` since the len of an empty set is also 0 (False)

Reply:
again partially mypy as for the copy, if you remove from a set during a for loop you will get an error because indices are removed while looping over the object. Here the one in `len` is unnecessary though so removing that.